engine/apps/api/views/organization.py

```diff
@@ -108,6 +109,28 @@ def get(self, request):
         return Response(code)
 
 
+class GetMattermostSetupDetails(APIView):
+    authentication_classes = (PluginAuthentication,)
+    permission_classes = (IsAuthenticated, RBACPermission)
+
+    rbac_permissions = {
+        "get": [RBACPermission.Permissions.INTEGRATIONS_WRITE],
```

Comment:
It would be nice to move this logic to the backend itself, maybe using the existing `GetChannelVerificationCode`? (although it could be confusing because in this case the code would be for a mattermost instance instead of a telegram or msteams channel), or defining a new optional messaging backend step for this. In an ideal world, it should be possible to write a messaging backend without changing any code in the other apps. Will give it some thought.

Reply:
I gave some thought. One way it could be is via polymorphic association we could have a single model to generate the tokens for each backend apps with a `type` field but that might be slightly complicated to unify that approach for the existing apps token implementation.